svm-transaction/src/svm_message.rs

```diff
@@ -60,6 +64,46 @@ pub trait SVMMessage: Debug {
         }
     }
 
+    /// If the message uses a durable nonce, return the pubkey of the nonce account
+    fn get_durable_nonce(&self) -> Option<&Pubkey> {
+        let account_keys = self.account_keys();
+        self.instructions_iter()
+            .nth(usize::from(NONCED_TX_MARKER_IX_INDEX))
+            .filter(
+                |ix| match account_keys.get(usize::from(ix.program_id_index)) {
+                    Some(program_id) => system_program::check_id(program_id),
+                    _ => false,
+                },
+            )
+            .filter(|ix| is_advance_nonce_account_instruction(ix.data))
```

Comment:
[df981ea](https://github.com/anza-xyz/agave/pull/3287/commits/df981ea24c6049603882f66168b28b5656e3e679)

Ended up moving all implementation into the main file, and then just converting the file to a more generic `tests`. Should be where the tests of all our default implementations for the traits go instead of specifically nonce-related stuff.